### ISSUE TITLE
pybids useful?

### ISSUE THREAD

**Issue by jcohenadad** [shimming-toolbox/shimming-toolbox#75]
Related to #35, there is already an actively-developed toolkit called [pybids](https://github.com/bids-standard/pybids), which might be useful for our BIDS-loader needs.

**Comment by gab-berest** [comment]
I'll check this to verify if we need to implement these functions for our specific usage

**Comment by gab-berest** [comment]
pybids seems to do a fairly good job at obtaining BIDS type data and parsing them to retrieve the correct run/session/subject needed. It will not load data as a 5D array when multiple echoes are found, but the data can be reshaped after loading it via pybids. Another problem is that it is not Windows supported for now. They say that most of the functions work on windows, but they don't support if there are malfunctions.
I have to say that maybe for parsing the BIDS data, it could be easier with this library and we could use it (once I've figured out exactly how it works, because it doesn't seem to work with data outputted by dcm2bids. Maybe a naming issue). We have to check if we want to live with the fact that Windows is not supported.

**Comment by jcohenadad** [comment]
> We have to check if we want to live with the fact that Windows is not supported.

we can definitely live with it, especially because WSL (and now WSL2) are getting increasingly reliable

**Comment by gab-berest** [comment]
In that case, it could be really usefull. I'll check to implement it for use in the load functions of the library.

**Comment by gab-berest** [comment]
Seems that, to load correctly the nifti files in the layout of PyBIDs, there is a need for tsv information files for each subject/session/etc. Now when I use dicom_to_nifit, the tsv files are note created. @gaspardcereza do youknow if I am missing something or a configuration that needs to be done to generate these tsv files for each subjetc, session, etc.?

**Comment by jcohenadad** [comment]
indeed tsv is part of bids convention— might be overkill to require that. or we could generate it with our dcm2bids wrapper

**Comment by gab-berest** [comment]
Yes, there is also a missing dataset_description.json file, participants.json and .tsv file that is correctly created with the MATLAB version of dicom_to_nifti, but not in the python version. Something to check.

**Comment by gab-berest** [comment]
I've created a new issue for the misisng files which prevent pybids from functionning [https://github.com/shimming-toolbox/shimming-toolbox-py/issues/77](url). @gaspardcereza Could you tell me why these steps where removed form the MATLAB version?
With these changes, PyBIDs is able to load the correct files. There is still a bug where it loads only the magnitude1 and magnitude2 files generated with the dcm2bids.json file provided. I'll dig deeper into it. I can't

**Comment by gaspardcereza** [comment]
The discussion about missing files has been moved [here](https://github.com/shimming-toolbox/shimming-toolbox-py/pull/78).

**Comment by gab-berest** [comment]
With dcm2bids, I can generate a sub-01 subject with files magnitude[1-6] with our config file. When I use PyBIDS, the magnitude[3-6] are not loaded and in the config file, it checks for a default name that can have only magnitude[1-2].

I couldn't find the answer in the bids documentation. Does magnitude greater than 2 exist in the bids convention?

**Comment by jcohenadad** [comment]
> Does magnitude greater than 2 exist in the bids convention?

hum... good question. This is quite uncommon (most ppl use 2 echoes). Is there some bids-hack we can do here?

**Comment by gab-berest** [comment]
> 
> 
> > Does magnitude greater than 2 exist in the bids convention?
> 
> hum... good question. This is quite uncommon (most ppl use 2 echoes). Is there some bids-hack we can do here?

I just checked and it is possible by sending him a config file that changes these line to increase the number of possible magnitude fields. We would need to keep another config file like the config file for dcm2bids.json however.

**Comment by 4rnaudB** [comment]
@po09i Can we close this one?
